workshops/pycuriosity/README.md

```diff
@@ -0,0 +1,417 @@
+```requests``` module to make a GET request to the API and ```argparse``` to validate command line arguments. Add the
+following imports using the [PEP-8 style guide](https://peps.python.org/pep-0008/#imports)
+```python
+import argparse
+
+import requests
+```
+- Next, you'll see the ```main()``` and ```explore_mars_rover_photos(file_name, date)``` functions, also with a bunch of
+TODOs. Notice how `explore_mars_rover_photos(file_name, date)`'s signature includes the requirements we need. We'll use
+`explore_mars_rover_photos(file_name, date)` to call the API and `main()` to sort out the command line arguments.
+- Finally, we have this code snippet:
+```python
+if __name__ == "__main__":
+    main()
+```
+If unfamiliar, this makes the program modular, so the functions defined can be used/imported in other projects.
+
+To access the Mars Rover Photos API, we need an API key(above). An API key is useful for authenticating data, so only
+trusted parties with the right API key can request information, which is very useful if the data costs money to collect,
+like weather data, population, etc. which might require paying for an API key. However, NASA's APIs are public and free,
+so we don't really need to signup and pay for an API key.
+
+But, declaring an API key in a program is dangerous(recall it's not in ```explore_mars_rover_photos(file_name, date)```'s signature), 
+as when code is visible to the public, everyone can copy-paste the API key and make requests, accessing sensitive data, which isn't great 
+if you had to pay for the API key, and it may lead to security breaches or data leaks. Even though we aren't paying for an API key, 
+and the data we're accessing is completely public and free of cost, as is convention, we'll export the API key as an environment variable 
+in the CLI(terminal/command line), and access it using the  builtin ```os``` module.
+
+
+### Setting up the ```os``` module
+To get an environment variable using the ```os``` module, we'll use ```os.environ.get()```. See the documentation
+[here](https://docs.python.org/3/library/os.html#file-names-command-line-arguments-and-environment-variables)
+```python
+import argparse
+import os
+
+import requests
+...
+API_KEY = os.environ.get("API_KEY")
+...
+```
+
+It's a good idea to get used to this, because this is standard practice when working with APIs.
+
+Next, we need to export the API key as an environment variable so the above code works. 
+To do so, click on Shell, next to Console, on replit
+
+![Clicking on shell](https://cloud-fx418po7s-hack-club-bot.vercel.app/0shell.png)
+
+If you signed up for an API key at [api.nasa.gov](https://api.nasa.gov/), copy it to your clipboard, and type in the 
+following command
+```shell
+export API_KEY=your_api_key
+```
+If you didn't sign up for an API key, NASA allows using a demo key, which can be used by
+```shell
+export API_KEY=DEMO_KEY
```

Comment:
it isn't immediately obvious that DEMO_KEY is the demo key haha but that might be because i didn't read it super in depth

Reply:
I can add that the DEMO_KEY is great if you don't want to sign up but has much lower request limits as per the website [api.nasa.gov](https://api.nasa.gov) to make it more clear

Reply:
hmm i thin it should be fine! once i read the documentation website i got it :)

Reply:
so no changes needed?

Reply:
nope :)